grails-app/services/org/pih/warehouse/data/LocationDataService.groovy

```diff
@@ -91,4 +101,8 @@ class LocationDataService {
         }
         return location
     }
+
+    Boolean isInternalOrZoneLocation(LocationTypeCode locationTypeCode) {
```

Comment:
I'd prefer here separate `isInternal` and `isZone`, on top of that these should be a methods in the LocationType. And then call `locationType?.isIntenal() || locationType?.isZone()` in the `if` condition.

Reply:
Done